# HG changeset patch
# User Mike Miller <mtmiller@octave.org>
# Date 1521768502 25200
#      Thu Mar 22 18:28:22 2018 -0700
# Branch stable
# Node ID 8033e53e6cb4f3129e5a85e7118f3916eccc4b72
# Parent  c4a3b8b3fbd86dcf8c0e32e5c4ed7a2d65b2427c
build: disallow building source distribution without Qt offscreen OpenGL (bug #53414)

* doc/interpreter/module.mk (doc-interpreter-dist-hook): Include the
HAVE_QT_OFFSCREEN preprocessor macro in the list of prerequisites to build the
source distribution.

diff --git a/doc/interpreter/module.mk b/doc/interpreter/module.mk
--- a/doc/interpreter/module.mk
+++ b/doc/interpreter/module.mk
@@ -300,16 +300,17 @@ doc_EXTRA_DIST += \
 
 # Prevent packaging of distribution unless all libraries
 # necessary to create documentation are present
 doc-interpreter-dist-hook:
 	@$(GREP) '#define HAVE_COLAMD 1' $(top_builddir)/config.h > /dev/null || { echo "Documentation creation requires missing COLAMD library.  Cannot package distribution!" ; exit 1; }
 	@$(GREP) '#define HAVE_CHOLMOD 1' $(top_builddir)/config.h > /dev/null || { echo "Documentation creation requires missing CHOLMOD library.  Cannot package distribution!" ; exit 1; }
 	@$(GREP) '#define HAVE_UMFPACK 1' $(top_builddir)/config.h > /dev/null || { echo "Documentation creation requires missing UMFPACK library.  Cannot package distribution!" ; exit 1; }
 	@$(GREP) '#define HAVE_QHULL 1' $(top_builddir)/config.h > /dev/null || { echo "Documentation creation requires missing QHULL library.  Cannot package distribution!" ; exit 1; }
+	@$(GREP) '#define HAVE_QT_OFFSCREEN 1' $(top_builddir)/config.h > /dev/null || { echo "Documentation creation requires Qt offscreen OpenGL rendering.  Cannot package distribution!" ; exit 1; }
 
 $(MUNGED_TEXI_SRC): $(DOCSTRING_FILES)
 
 ## These two texi files have an additional dependency through the
 ## @EXAMPLEFILE macro.
 %reldir%/oop.texi: $(examples_code_SRC)
 %reldir%/external.texi: $(examples_code_SRC)
 
